oneDNN 3.0.1 uses non public headers from ACL

# Summary
oneDNN 3.0.1 uses non public headers from ACL. So, building oneDNN 3.0.1 against `ComputeLibrary-devel` package leads to the following error:
```
[ 1131s] [ 16%] Building CXX object src/cpu/CMakeFiles/dnnl_cpu.dir/platform.cpp.o
[ 1131s] cd /home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/build/src/cpu && /usr/bin/c++ -DDNNL_AARCH64=1 -DDNNL_AARCH64_USE_ACL -DDNNL_DLL -DDNNL_DLL_EXPORTS -DDNNL_ENABLE_CPU_ISA_HINTS -DDNNL_ENABLE_ITT_TASKS -DDNNL_ENABLE_MAX_CPU_ISA -D__STDC_CONSTANT_MACROS -D__STDC_LIMIT_MACROS -I/home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/include -I/home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/build/include -I/home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/src -mbranch-protection=standard -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fopenmp -fvisibility-inlines-hidden  -Wall -Wno-unknown-pragmas -fvisibility=internal -O3 -mcpu=native  -fPIC -Wformat -Wformat-security -fstack-protector-strong  -Wmissing-field-initializers -Wmissing-field-initializers  -Wno-strict-overflow -Wno-maybe-uninitialized  -O2 -g -DNDEBUG -std=c++14 -MD -MT src/cpu/CMakeFiles/dnnl_cpu.dir/platform.cpp.o -MF CMakeFiles/dnnl_cpu.dir/platform.cpp.o.d -o CMakeFiles/dnnl_cpu.dir/platform.cpp.o -c /home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/src/cpu/platform.cpp
[ 1131s] /home/abuild/rpmbuild/BUILD/oneDNN-3.0.1/src/cpu/platform.cpp:41:10: fatal error: src/common/cpuinfo/CpuInfo.h: No such file or directory
[ 1131s]    41 | #include "src/common/cpuinfo/CpuInfo.h"
[ 1131s]       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 1131s] compilation terminated.
```

# Version
3.0.1

# Environment
* CPU make and model: aarch64
* OS version (`uname -a`): Linux hctw01 6.2.1-1-default #1 SMP PREEMPT_DYNAMIC Mon Feb 27 11:39:51 UTC 2023 (69e0e95) aarch64 aarch64 aarch64 GNU/Linux
* Compiler version (`gcc --version`) 12
* CMake version (`cmake --version`) 3.25
* CMake output log
* git hash (`git log -1 --format=%H`)

# Steps to reproduce
Try to build oneDNN on openSUSE Tumbleweed

# Observed behavior
Build fails with ACL on aarch64.

# Expected behavior
Build should succeed.

--- a/cmake/FindACL.cmake
+++ b/cmake/FindACL.cmake
@@ -24,14 +24,12 @@
 find_path(ACL_INCLUDE_DIR
   NAMES arm_compute/graph.h
   PATHS ENV ACL_ROOT_DIR
-  NO_DEFAULT_PATH
   )
 
 find_library(ACL_LIBRARY
   NAMES arm_compute
   PATHS ENV ACL_ROOT_DIR
-  PATH_SUFFIXES build
-  NO_DEFAULT_PATH
+  PATH_SUFFIXES lib build
   )
 
 include(FindPackageHandleStandardArgs)
@@ -56,7 +54,7 @@ if(ACL_FOUND)
   find_library(ACL_GRAPH_LIBRARY
     NAMES arm_compute_graph
     PATHS ENV ACL_ROOT_DIR
-    PATH_SUFFIXES build
+    PATH_SUFFIXES lib build
     )
 
   list(APPEND ACL_INCLUDE_DIRS

--- a/src/cpu/platform.cpp
+++ b/src/cpu/platform.cpp
@@ -1,7 +1,7 @@
 /*******************************************************************************
 * Copyright 2020-2023 Intel Corporation
 * Copyright 2020 FUJITSU LIMITED
-* Copyright 2022 Arm Ltd. and affiliates
+* Copyright 2022-2024 Arm Ltd. and affiliates
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
@@ -37,8 +37,6 @@
 #if DNNL_AARCH64_USE_ACL
 // For checking if fp16 isa is supported on the platform
 #include "arm_compute/core/CPP/CPPTypes.h"
-// For setting the number of threads for ACL
-#include "src/common/cpuinfo/CpuInfo.h"
 #endif
 #endif
 
@@ -208,7 +206,7 @@ unsigned get_num_cores() {
 #if DNNL_X64
     return x64::cpu().getNumCores(Xbyak::util::CoreLevel);
 #elif DNNL_AARCH64_USE_ACL
-    return arm_compute::cpuinfo::num_threads_hint();
+    return aarch64::cpu().getNumCores(Xbyak_aarch64::util::CoreLevel);
 #else
     return 1;
 #endif

